### ISSUE TITLE
`hideOthers` hides everything when target is in Shadow DOM and paren is not explicitly provided

### ISSUE THREAD

**Issue by harunurhan** [theKashey/aria-hidden#14]
Traversal to mark all parent chain of as "elementToKeep" gets stuck at shadow root since it's using `element.parentNode`, so no top level parent (direct child of `body`) is reached.
https://github.com/theKashey/aria-hidden/blob/95843f2f35c5abb413835529b6bc6bfd4fa93a99/src/index.ts#L50

And since it's using `target.ownerDocument.body` to start setting `aria-hidden: "true"`, it just stops after hiding all of its children, which is pretty much everything on the page :(
https://github.com/theKashey/aria-hidden/blob/95843f2f35c5abb413835529b6bc6bfd4fa93a99/src/index.ts#L10

Consumer can indeed address this issue manually, but maybe we could change the default `getDefaultParent` to return shadow root to make this easier for everyone, since not all will notice this is broken.
https://github.com/theKashey/aria-hidden/blob/95843f2f35c5abb413835529b6bc6bfd4fa93a99/src/index.ts#L3

Maybe something like that could work?

```js
const rootNode = element.getRootNode();
const isShadowRoot = rootNode instanceof ShadowRoot;
return isShadowRoot ? rootNode : element.owerDocument.body;
```

However in some cases if not most, I don't think hiding everything only within a shadow DOM is going to work, so it could be better to fix parent traversal.

```ts
keep(el.parentNode ?? el.host);
```

**Comment by theKashey** [comment]
- check if parent contains target
- if not
  - if there is owner document - move target there
    - repeat
  - display warning. If parent equal to the body - do nothing (we really don't want to "hide" everything)

This should "bubble up" target to current document, po

**Comment by theKashey** [comment]
Resolved in v1.2.2
